### PR TITLE
builder: implement orientation according to coordinates

### PR DESCRIPTION
**Pull request by Tristramg**
This should be a pretty common problem for any user of the library, so it makes sens to have this in the library

This allowed to find an bug, hence the first tests in the builder (ref. #57)

### CODE REVIEW

File: src/builder.rs [hunk 1]
```diff
@@ -506,4 +506,112 @@ impl<'fbb> Builder<'fbb> {
...
+            segment_index: s1,
+            reversed: true,
+        };
+        let sot2 = super::SegmentOfTraversal {
+            segment_index: s2,
+            reversed: false,
+        };
+        let traversal = b.add_traversal("traversal", &[sot1, sot2]);
+        assert_eq!(b.nodes_of_traversal[traversal], [11, 10, 12]);
+    }
```
Comment (Castavo) [inline src/builder.rs:571]: Why not put this in its own test ?

Comment (Castavo) [inline src/builder.rs:571]: Why not put this in its own test ?

File: src/builder.rs [hunk 2]
```diff
@@ -506,4 +506,112 @@ impl<'fbb> Builder<'fbb> {
...
+    #[test]
+    fn orientation_on_curve() {
+        let mut b = Builder::new();
+        let traversal = build_traversal(&mut b);
+
+        let point_a = point! {x: 1., y: 1.};
+        let point_b = point! {x: 2., y: -1.};
+
+        b.orient_along_points(traversal, point_a, point_b).unwrap();
+        assert_eq!(b.temp_traversal[0].segments[0].segment_index, 0);
```
Comment (Castavo) [inline src/builder.rs:582]: Maybe 
```suggestion
        assert_eq!(b.temp_traversal[traversal].segments[0].segment_index, 0);
```
is more maintainable ? (not sure)

File: src/builder.rs [hunk 3]
```diff
@@ -506,4 +506,112 @@ impl<'fbb> Builder<'fbb> {
...
+        let mut b = Builder::new();
+        let traversal = build_traversal(&mut b);
+
+        // a-b---[the curve]
+        let point_a = point! {x: -1., y: 0.};
+        let point_b = point! {x: -2., y: 0.};
+        b.orient_along_points(traversal, point_a, point_b).unwrap();
+        assert_eq!(b.temp_traversal[traversal].segments[0].segment_index, 0);
+        b.orient_along_points(traversal, point_b, point_a).unwrap();
+        assert_eq!(b.temp_traversal[traversal].segments[0].segment_index, 1);
```
Comment (Castavo) [inline src/builder.rs:601]: Why not check b.nodes_of_traversal here ?

**Review by Castavo — APPROVED** [review]
The tests run
I have a few comment but it's nitpicking

File: src/builder.rs (hunk 2 above)
Comment (Castavo) [reply src/builder.rs:582]: At least it would be more coherent with the other tests 😛

File: src/builder.rs (hunk 1 above)
Comment (Tristramg) [reply src/builder.rs:571]: The idea is to have to test if the behavior really changes: the nominal case, and the variation

File: src/builder.rs (hunk 3 above)
Comment (Tristramg) [reply src/builder.rs:601]: In the first test we want to test that everything has been reversed

here we want to test some edge case behavior when using points that project to an extremity of the curve

Comment (Castavo) [reply src/builder.rs:601]: Okay, fair enough
